data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostPrepareBeaconProposer.java

```diff
@@ -0,0 +1,92 @@
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.ValidatorDataProvider;
+import tech.pegasys.teku.api.schema.merge.BeaconPreparableProposer;
+import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.infrastructure.http.HttpStatusCodes;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class PostPrepareBeaconProposer extends AbstractHandler implements Handler {
+  public static final String ROUTE = "/eth/v1/validator/prepare_beacon_proposer";
+  private static final Logger LOG = LogManager.getLogger();
+
+  public PostPrepareBeaconProposer(final DataProvider provider, final JsonProvider jsonProvider) {
+    this(provider.getValidatorDataProvider(), jsonProvider);
+  }
+
+  public PostPrepareBeaconProposer(
+      final ValidatorDataProvider provider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.POST,
+      summary = "Provide beacon node with proposals for the given validators.",
+      tags = {TAG_VALIDATOR, TAG_VALIDATOR_REQUIRED},
+      requestBody =
+          @OpenApiRequestBody(content = {@OpenApiContent(from = BeaconPreparableProposer[].class)}),
+      description =
+          "Prepares the beacon node for potential proposers by supplying information required when proposing blocks for the given validators. The information supplied for each validator index is considered persistent until overwritten by new information for the given validator index, or until the beacon node restarts.\n\n"
+              + "Note that because the information is not persistent across beacon node restarts it is recommended that either the beacon node is monitored for restarts or this information is refreshed by resending this request periodically (for example, each epoch).\n\n"
+              + "Also note that requests containing currently inactive or unknown validator indices will be accepted, as they may become active at a later epoch.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            description = "Preparation information has been received."),
+        @OpenApiResponse(status = RES_BAD_REQUEST, description = "Invalid parameter supplied."),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR, description = "Beacon node internal error.")
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    try {
+      final BeaconPreparableProposer[] request =
+          parseRequestBody(ctx.body(), BeaconPreparableProposer[].class);
+
+      LOG.trace(
+          "received: {}",
+          Arrays.stream(request)
+              .map(BeaconPreparableProposer::toString)
+              .collect(Collectors.joining(",")));
```

Comment:
This will wind up creating the string on every invocation even though it won't be logged. You could pass a `Supplier` to avoid that but I'm not sure it's a particularly useful log to have in the prod code anyway so would just remove it.

Reply:
right. BTW logging will go away in the coming PRs